x/metadata/types/signer_utils_test.go

```diff
@@ -0,0 +1,2470 @@
+		})
+	}
+}
+
+func TestPartyDetails_Copy(t *testing.T) {
+	tests := []struct {
+		name string
+		pd   *PartyDetails
+	}{
+		{
+			name: "nil",
+			pd:   nil,
+		},
+		{
+			name: "empty",
+			pd:   &PartyDetails{},
+		},
+		{
+			name: "just address",
+			pd:   &PartyDetails{address: "address_value"},
+		},
+		{
+			name: "just role",
+			pd:   &PartyDetails{role: PartyType_PARTY_TYPE_OMNIBUS},
+		},
+		{
+			name: "just optional",
+			pd:   &PartyDetails{optional: true},
+		},
+		{
+			name: "just acc",
+			pd:   &PartyDetails{acc: sdk.AccAddress("acc_value")},
+		},
+		{
+			name: "just signer",
+			pd:   &PartyDetails{signer: "signer_value"},
+		},
+		{
+			name: "just signerAcc",
+			pd:   &PartyDetails{signerAcc: sdk.AccAddress("signerAcc_value")},
+		},
+		{
+			name: "just canBeUsedBySpec",
+			pd:   &PartyDetails{canBeUsedBySpec: true},
+		},
+		{
+			name: "just usedBySpec",
+			pd:   &PartyDetails{usedBySpec: true},
+		},
+		{
+			name: "required party",
+			pd: WrapRequiredParty(Party{
+				Address:  "required_address",
+				Role:     PartyType_PARTY_TYPE_CUSTODIAN,
+				Optional: false,
+			}),
+		},
+		{
+			name: "available party",
+			pd: WrapAvailableParty(Party{
+				Address:  "available_address",
+				Role:     PartyType_PARTY_TYPE_ORIGINATOR,
+				Optional: true,
+			}),
+		},
+		{
+			name: "everything populated",
+			pd: &PartyDetails{
+				address:         "another_address",
+				role:            PartyType_PARTY_TYPE_AFFILIATE,
+				optional:        true,
+				acc:             sdk.AccAddress("the_acc_field"),
+				signer:          "another_signer",
+				signerAcc:       sdk.AccAddress("the_signerAcc_field"),
+				canBeUsedBySpec: true,
+				usedBySpec:      true,
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var actual *PartyDetails
+			testFunc := func() {
+				actual = tc.pd.Copy()
+			}
+			require.NotPanics(t, testFunc, "Copy()")
+			require.Equal(t, tc.pd, actual, "result of Copy()")
+			if tc.pd != nil && actual != nil {
+				assert.NotSame(t, tc.pd, actual, "result of Copy()")
+				assert.NotSame(t, tc.pd.acc, actual.acc, "acc field")
+				if len(actual.acc) > 0 {
+					// Change the first byte in the copy to make sure it doesn't also change in the original.
+					actual.acc[0] = actual.acc[0] + 1
+					assert.NotEqual(t, tc.pd.acc, actual.acc, "the acc field after changing it in the copy")
+					// And put it back so we don't mess up anything else.
+					actual.acc[0] = actual.acc[0] - 1
+				}
+				assert.NotSame(t, tc.pd.signerAcc, actual.signerAcc, "signerAcc field")
+				if len(actual.signerAcc) > 0 {
+					// Change the first byte in the copy to make sure it doesn't also change in the original.
+					actual.signerAcc[0] = actual.signerAcc[0] + 1
+					assert.NotEqual(t, tc.pd.signerAcc, actual.signerAcc, "the signerAcc field after changing it in the copy")
+					actual.signerAcc[0] = actual.signerAcc[0] - 1
+				}
```

Comment:
A `byte` is just a `uint8`. An overflow just wraps it around, but it's still changed and that's what really matters. And using bitmath is generally frowned upon because it's harder to read and understand.